File: answers/leetcode/804. Unique Morse Code Words/Solution.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        MORSE = [".-","-...","-.-.","-..",".","..-.","--.",
                 "....","..",".---","-.-",".-..","--","-.",
                 "---",".--.","--.-",".-.","...","-","..-",
                 "...-",".--","-..-","-.--","--.."]
        return len({"".join(MORSE[ord(char) - ord('a')] for char in word) for word in words})

File: answers/leetcode/804. Unique Morse Code Words/Solution.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] MORSE = new String[]{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> representations = new HashSet<String>();
        for (String word : words) {
            StringBuilder sb = new StringBuilder();
            for (char c : word.toCharArray()) {
                int index = c - 'a';
                sb.append(MORSE[index]);
            }
            representations.add(sb.toString());
        }
        return representations.size();
    }
}